### PR TITLE
Rwc2019 robot skills

### PR DESCRIPTION
**Pull request by KevinDang74**

### CODE REVIEW

File: robot_skills/src/robot_skills/api.py [hunk 1]
```diff
@@ -34,11 +38,13 @@ def query(self, description, grammar, target, timeout=10):
         :param target: string identifying the target of the grammar to recognize
         :param timeout: timeout in seconds (float)
         """
+        rospy.sleep(2.0)  # Check if this reduces the speech bug
```
Comment (DvanDinther) [inline robot_skills/src/robot_skills/api.py]: Should this still be here?

File: robot_skills/src/robot_skills/arms.py [hunk 2]
```diff
@@ -696,15 +749,19 @@ def _send_joint_trajectory(self, joints_references, timeout=rospy.Duration(5), j
...
         for joints_reference in joints_references:
+            max_diff = sorted([abs(prev - new) for prev, new in zip(prev_joint_ref, joints_reference)])[-1]
             if len(joints_reference) != len(joint_names):
                 rospy.logwarn('Please use the correct %d number of joint references (current = %d'
                               % (len(joint_names), len(joints_references)))
+            time_from_start += max_diff/max_joint_vel
             ps.append(JointTrajectoryPoint(
                 positions=joints_reference,
-                time_from_start=time_from_start))
+                time_from_start=rospy.Duration.from_sec(time_from_start)))
```
Comment (DvanDinther) [inline robot_skills/src/robot_skills/arms.py]: If I understand it correctly, the above code uses the max joint velocity along with relative position difference per joint to calculate `time_from_start`. Why not use ros time to track time instead of approximating it this way?

File: robot_skills/src/robot_skills/base.py [hunk 3]
```diff
@@ -193,26 +198,37 @@ def move(self, position_constraint_string, frame):
...
+        :param ay: sideways acceleration in m/s^2
+        :param ath: rotational acceleration in rad/s^2
         :param timeout: duration for this motion in seconds
         """
         # Cancel the local planner goal
         self.local_planner.cancelCurrentPlan()
 
         v = geometry_msgs.msg.Twist()        # Initialize velocity
         t_start = rospy.Time.now()
+        t_end = rospy.Time.now() + rospy.Duration.from_sec(timeout)
```
Comment (DvanDinther) [inline robot_skills/src/robot_skills/base.py]: `t_end = t_start + rospy.Duration.from_sec(timeout)`

File: robot_skills/src/robot_skills/base.py [hunk 4]
```diff
@@ -193,26 +198,37 @@ def move(self, position_constraint_string, frame):
...
+
+        def _abs_max(value, abs_max_value):
+            if value == 0:
+                return 0
+            return (value / abs(value)) * min(abs_max_value, abs(value))
```
Comment (DvanDinther) [inline robot_skills/src/robot_skills/base.py]: `(value / abs(value))` determines the sign of `value` so maybe numpy.sign() is a better alternative so that there is no need for the `if`? Although numpy isn't in here yet. 

Moreover, is the abs_max_value really absolute? Depends on use, but seems to me like another `abs` function should be used.

File: robot_skills/src/robot_skills/force_sensor.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    pass
+
+
+class ForceSensor(object):
+    def __init__(self, topic, force_norm_threshold=2.5):
+        """
+        Class for conveniently using a force sensor
+
+        :param topic: Topic to use for WrenchStamped measurement
+        :param force_norm_threshold: Edge up if norm exceeds this value (defaults to 2.5)
```
Comment (DvanDinther) [inline robot_skills/src/robot_skills/force_sensor.py]: 2.5 what? Newton?

File: robot_skills/src/robot_skills/force_sensor.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        """
+        self._topic = topic
+
+        self._calibrated_msg = None
+        self._edge_up = False
+        self._force_norm_threshold = force_norm_threshold
+
+    def _wrench_callback(self, msg):
```
Comment (DvanDinther) [inline robot_skills/src/robot_skills/force_sensor.py]: docstrings

File: robot_skills/src/robot_skills/api.py (hunk 1 above)
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/api.py]: This should be removed

File: robot_skills/src/robot_skills/api.py [hunk 7]
```diff
@@ -50,6 +56,26 @@ def query(self, description, grammar, target, timeout=10):
 
         return answer
 
+    def show_image(self, path_to_image, seconds=5.0):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/api.py]: Share code between `show_image` and `show_image_from_msg`.

File: robot_skills/src/robot_skills/api.py [hunk 8]
```diff
@@ -50,6 +56,26 @@ def query(self, description, grammar, target, timeout=10):
...
+        """
+        Show an image on the HMI display interface
+        :param path_to_image: Absolute path to image file
+        :param seconds: How many seconds you would like to display the image on the screen
+        """
+        compressed_image_msg = CvBridge().cv2_to_compressed_imgmsg(cv2.imread(path_to_image))
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/api.py]: Is the creation of a new CvBridge instance each time the most efficient way?

File: robot_skills/src/robot_skills/arms.py [hunk 9]
```diff
@@ -597,13 +647,14 @@ def occupied_by(self, value):
...
 
-    def send_gripper_goal(self, state, timeout=5.0):
+    def send_gripper_goal(self, state, timeout=5.0, max_torque=0.1):
         """
         Send a GripperCommand to the gripper of this arm and wait for finishing
         :param state: open or close
         :type state: str (GripperState)
         :param timeout: timeout in seconds; timeout of 0.0 is not allowed
         :type timeout: float
+        :param max_torque: How much force to apply, range between 0 and 1, only applied if closing the gripper
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: Torque/force, please be consistent. Also the range between 0 and 1 is very unclear. Please provide units.

File: robot_skills/src/robot_skills/base.py (hunk 3 above)
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/base.py]:
```suggestion
        t_end = t_start + rospy.Duration.from_sec(timeout)
```

File: robot_skills/src/robot_skills/force_sensor.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import numpy as np
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/force_sensor.py]: Only import the required module of numpy.

File: robot_skills/src/robot_skills/mockbot.py [hunk 11]
```diff
@@ -394,7 +389,9 @@ def hear(text):
         pub.publish(std_msgs.msg.String(text))
 
     def save_sentence(sentence):
-        """Let mockbot say a sentence and them move the generated speech file to a separate file that will not be overwritten"""
+        """
+        Let mockbot say a sentence and them move the generated speech file to a separate file that will not be overwritten
+        """
         speak(sentence)
         import os
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/mockbot.py]: Good to fix this import.

File: robot_skills/src/robot_skills/perception.py [hunk 12]
```diff
@@ -48,6 +49,22 @@ def __init__(self, robot_name, tf_listener, image_topic=None, projection_srv=Non
...
         self._projection_srv = self.create_service_client(projection_srv_name, Project2DTo3D)
+        self._person_recognition_3d_srv = \
+            self.create_service_client('/' + robot_name + '/people_recognition/detect_people_3d', RecognizePeople3D)
+
+        # camera topics
+        self.depth_info_sub = message_filters.Subscriber('{}/depth_registered/camera_info'.format(self._camera_base_ns), CameraInfo)
+        self.depth_sub = message_filters.Subscriber('{}/depth_registered/image'.format(self._camera_base_ns), Image)
+        self.rgb_sub = message_filters.Subscriber('{}/rgb/image_raw'.format(self._camera_base_ns), Image)
+
+        ts = message_filters.ApproximateTimeSynchronizer([self.rgb_sub, self.depth_sub, self.depth_info_sub],
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/perception.py]: Make this a member variable, just like the subs?

File: robot_skills/src/robot_skills/robot.py [hunk 13]
```diff
@@ -41,11 +47,73 @@ def __init__(self, robot_name="", wait_services=False):
...
+            visualization_msgs.msg.MarkerArray,
+            queue_size=1,
+        )
+
         # Check hardware status
         self._hardware_status_sub = rospy.Subscriber("/" + self.robot_name + "/hardware_status", DiagnosticArray, self.handle_hardware_status)
 
         self.laser_topic = "/"+self.robot_name+"/base_laser/scan"
 
+    def publish_rack(self):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/robot.py]: RoboCup hack

File: robot_skills/src/robot_skills/util/entity.py [hunk 14]
```diff
@@ -239,11 +241,9 @@ def from_entity_info(e):
     if e.type == 'person':
         try:
             pp_dict = yaml.load(e.data)
-            tags_dict = dict(zip(pp_dict['tagnames'], pp_dict['tags']))
             del pp_dict['position']
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/util/entity.py]: Why the deletion of data from the dict? Do we need to sync the person properties to (new) message definitions?

File: robot_skills/test.py [hunk 15]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+from ed_sensor_integration.srv import RayTrace
+from geometry_msgs.msg import PoseStamped
+from robot_skills import perception
+import rospy
+from std_msgs.msg import Header
+from tf import TransformListener
+
+rospy.init_node('dsada')
```
Comment (MatthijsBurgh) [inline robot_skills/test.py]: Normal node name please

File: robot_skills/test.py [hunk 16]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
```
Comment (MatthijsBurgh) [inline robot_skills/test.py]: Filename is too generic

File: robot_skills/test.py [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+rospy.init_node('dsada')
+
+p = perception.Perception("hero", None, None, camera_base_ns="hero/head_rgbd_sensor")
+
+srv_proxy = rospy.ServiceProxy("/hero/ed/ray_trace", RayTrace)
+
+tf_listener = TransformListener()
+
+while not rospy.is_shutdown():
```
Comment (MatthijsBurgh) [inline robot_skills/test.py]: Unclear what is being tested in this file. No documentation, generic filename, etc. Also this looks like code for standalone usage, not a real test. (For in CI)

File: robot_skills/test/test_force_sensor_edge_up [hunk 18]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
```
Comment (MatthijsBurgh) [inline robot_skills/test/test_force_sensor_edge_up]: - Filename extension would be nice
- hardware tests should be seperated from CI tests,

File: robot_skills/test/test_move_down_until_force_sensor_edge_up [hunk 19]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
```
Comment (MatthijsBurgh) [inline robot_skills/test/test_move_down_until_force_sensor_edge_up]: - Filename extension would be nice
- hardware tests should be seperated from CI tests,

File: robot_skills/src/robot_skills/arms.py [hunk 20]
```diff
@@ -194,6 +198,44 @@ def base_offset(self):
         """
         return self._arm.base_offset
 
+    def move_down_until_force_sensor_edge_up(self, timeout=10, retract_distance=0.01):
+        """
+        Move down the arm (hero specific, only joint arm_lift_joint) until the force sensor detects an edge up
```
Comment (Rayman) [inline robot_skills/src/robot_skills/arms.py]: `robot_skills` are robot independent, so this funcionality belongs somewhere else.

File: robot_skills/src/robot_skills/ebutton.py [hunk 21]
```diff
@@ -7,16 +7,21 @@
 
 
 class EButton(RobotPart):
-    """
-    Interface to amigo emergency switch. Listens to /emergency_switch topic
-    """
+    def __init__(self, robot_name, tf_listener, topic=None):
+        """
+        Interface to amigo emergency switch. Listens to /emergency_switch topic
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/ebutton.py]: I think the behaviour of HERO is inversed compared to AMIGO and SERGIO, so this should be checked and incorperated.

File: robot_skills/test/lazy_subscriber_test.py [hunk 22]
```diff
@@ -0,0 +1,19 @@
+# ROS
```
Comment (MatthijsBurgh) [inline robot_skills/test/lazy_subscriber_test.py]: This should be removed from the test folder and move somewhere else. The test folder will be used for CI tests only.

File: robot_skills/src/robot_skills/arms.py (hunk 2 above)
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/arms.py]: `JointTrajectoryPoint` is a ROS message that has a `rospy.Duration` as the `time_from_start` field, assigning an `int` won't work

File: robot_skills/src/robot_skills/force_sensor.py (hunk 5 above)
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/force_sensor.py]: You sound just like my Physics teacher. "What are those? Newtons? Joules? Bananas?"

File: robot_skills/src/robot_skills/api.py (hunk 8 above)
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/api.py]: Certainly not, but given the frequency I don't consider it an issue. Would be nice, but there's more important stuff

File: robot_skills/src/robot_skills/force_sensor.py (hunk 10 above)
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/force_sensor.py]: Import as np is the convention used all over the internet.

File: robot_skills/src/robot_skills/torso.py [hunk 23]
```diff
@@ -24,6 +21,7 @@ def __init__(self, robot_name, tf_listener):
         super(Torso, self).__init__(robot_name=robot_name, tf_listener=tf_listener)
 
         self.joint_names = self.load_param('skills/torso/joint_names')
+        self._arm_joint_names = self.load_param('skills/arm/joint_names')
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/torso.py]: HACK TO THE MAX :cry:

File: robot_skills/src/robot_skills/force_sensor.py (hunk 10 above)
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/force_sensor.py]: Yep the AS is correct. But why import whole numpy if you just need one function/class?

File: robot_skills/src/robot_skills/ebutton.py (hunk 21 above)
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/ebutton.py]: Nope, same behaviour

File: robot_skills/src/robot_skills/arms.py [hunk 24]
```diff
@@ -101,10 +102,11 @@ def has_joint_goal(self, configuration):
         """
         return configuration in self._available_joint_goals
 
-    def send_joint_goal(self, configuration, timeout=5.0):
+    def send_joint_goal(self, configuration, timeout=5.0, max_joint_vel=0.7):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 25]
```diff
@@ -113,10 +115,11 @@ def has_joint_trajectory(self, configuration):
         """
         return configuration in self._available_joint_trajectories
 
-    def send_joint_trajectory(self, configuration, timeout=5):
+    def send_joint_trajectory(self, configuration, timeout=5, max_joint_vel=0.7):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 26]
```diff
@@ -542,31 +595,35 @@ def send_goal(self, frameStamped, timeout=30, pre_grasp=False, first_joint_pos_o
                 rospy.logerr('grasp precompute goal failed: \n%s', repr(myargs))
                 return False
 
-    def send_joint_goal(self, configuration, timeout=5.0):
+    def send_joint_goal(self, configuration, timeout=5.0, max_joint_vel=0.7):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 27]
```diff
@@ -542,31 +595,35 @@ def send_goal(self, frameStamped, timeout=30, pre_grasp=False, first_joint_pos_o
...
             return self._send_joint_trajectory([self.default_configurations[configuration]],
-                                               timeout=rospy.Duration(timeout))
+                                               timeout=rospy.Duration.from_sec(timeout),
+                                               max_joint_vel=max_joint_vel)
         else:
             rospy.logwarn('Default configuration {0} does not exist'.format(configuration))
             return False
 
-    def send_joint_trajectory(self, configuration, timeout=5):
+    def send_joint_trajectory(self, configuration, timeout=5.0, max_joint_vel=0.7):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 28]
```diff
@@ -674,7 +733,7 @@ def handover_to_robot(self, timeout=10):
             rospy.logerr(e)
             return False
 
-    def _send_joint_trajectory(self, joints_references, timeout=rospy.Duration(5), joint_names = None):
+    def _send_joint_trajectory(self, joints_references, max_joint_vel=0.7, timeout=rospy.Duration(5), joint_names=None):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 29]
```diff
@@ -875,13 +939,13 @@ def send_goal(self, frameStamped, timeout=30, pre_grasp=False, first_joint_pos_o
                   allowed_touch_objects=[]):
         return False
 
-    def send_joint_goal(self, configuration, timeout=5.0):
+    def send_joint_goal(self, configuration, max_joint_vel=0.7, timeout=5.0):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 30]
```diff
@@ -875,13 +939,13 @@ def send_goal(self, frameStamped, timeout=30, pre_grasp=False, first_joint_pos_o
...
         return False
 
-    def send_joint_trajectory(self, configuration, timeout=5):
+    def send_joint_trajectory(self, configuration, max_joint_vel=0.7, timeout=5):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/arms.py [hunk 31]
```diff
@@ -875,13 +939,13 @@ def send_goal(self, frameStamped, timeout=30, pre_grasp=False, first_joint_pos_o
...
         return False
 
-    def _send_joint_trajectory(self, joints_references, timeout=rospy.Duration(5), joint_names = None):
+    def _send_joint_trajectory(self, joints_references, max_joint_vel=0.7, timeout=rospy.Duration(5), joint_names = None):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: max_joint_vel should be a list

File: robot_skills/src/robot_skills/base.py [hunk 32]
```diff
@@ -193,26 +199,37 @@ def move(self, position_constraint_string, frame):
...
         :param timeout: duration for this motion in seconds
         """
         # Cancel the local planner goal
         self.local_planner.cancelCurrentPlan()
 
         v = geometry_msgs.msg.Twist()        # Initialize velocity
         t_start = rospy.Time.now()
+        t_end = t_start + rospy.Duration.from_sec(timeout)
+
+        def _abs_max(value, abs_max_value):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/base.py]: This is clamping right?

File: robot_skills/src/robot_skills/force_sensor.py [hunk 33]
```diff
@@ -0,0 +1,57 @@
+from numpy import array as np_array
+from numpy.linalg import norm as np_norm
+import rospy
+from geometry_msgs.msg import WrenchStamped
+
+
+class TimeOutException(Exception):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/force_sensor.py]: Very generic, could maybe be moved to a more generic place.

File: robot_skills/src/robot_skills/force_sensor.py [hunk 34]
```diff
@@ -0,0 +1,57 @@
...
+    pass
+
+
+class ForceSensor(object):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/force_sensor.py]: Would prefer if this is a robot_part

File: robot_skills/src/robot_skills/torso.py [hunk 35]
```diff
@@ -94,8 +91,9 @@ def _send_goal(self, torso_pos, timeout=0.0, tolerance=[]):
 
         torso_goal = control_msgs.msg.FollowJointTrajectoryGoal()
         torso_goal_point = trajectory_msgs.msg.JointTrajectoryPoint()
-        torso_goal.trajectory.joint_names = self.joint_names
+        torso_goal.trajectory.joint_names = list(self.joint_names)
         torso_goal_point.positions = torso_pos
+        torso_goal_point.time_from_start = rospy.Duration.from_sec(1.5)
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/torso.py]: Magic number

File: robot_skills/src/robot_skills/util/entity.py [hunk 36]
```diff
@@ -160,7 +160,7 @@ def __repr__(self):
 
 class PersonProperties(object):
     def __init__(self, name, age, emotion, gender, gender_confidence, pointing_pose, posture, reliability, shirt_colors,
-                 tags_dict, velocity, parent_entity):
+                 tags, tagnames, velocity, parent_entity):
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/util/entity.py]: Why the change? Would always let keys become before values.

File: robot_skills/test/test_move_down_until_force_sensor_edge_up.py [hunk 37]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
```
Comment (MatthijsBurgh) [inline robot_skills/test/test_move_down_until_force_sensor_edge_up.py]: Should me moved to somewhere else, probably examples

File: robot_skills/src/robot_skills/util/entity.py (hunk 14 above)
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/util/entity.py]: Position is a property of the Entity, not of Entity.PersonProperties

File: robot_skills/src/robot_skills/base.py [hunk 38]
```diff
@@ -193,26 +199,37 @@ def move(self, position_constraint_string, frame):
 
         return plan
 
-    def force_drive(self, vx, vy, vth, timeout):
+    def force_drive(self, vx, vy, vth, timeout, ax=float('inf'), ay=float('inf'), ath=float('inf')):
```
[the remark this answers is not in the document]
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/base.py]: If it works, don't touch it

File: robot_skills/src/robot_skills/arms.py [hunk 39]
```diff
@@ -146,6 +149,7 @@ def send_gripper_goal(self, state, timeout=5.0, gripper_type=None):
         :type timeout: float
         :param gripper_type: Optional type of gripper to perform the action.
         :type gripper_type: str
+        :param max_torque: How much force to apply, range between 0 and 1, only applied if closing the gripper
```
[the remark this answers is not in the document]
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/arms.py]: This max_torque is passed through this call stack

1. https://github.com/tue-robotics/hero_bridge/blob/master/src/gripper_controller/gripper_controller.py#L38 
2. https://github.com/tue-robotics/hero_bridge/blob/master/src/gripper_controller/gripper_controller.py#L48
3. https://github.com/tue-robotics/hero_bridge/blob/master/src/gripper_controller/gripper_controller.py#L58
4. https://github.com/tue-robotics/hero_bridge/blob/master/src/gripper_controller/gripper_controller.py#L85

The max _force_ is 40N according to https://docs.hsr.io/hsrb_user_manual_en/overview/basic_specifications.html

File: robot_skills/src/robot_skills/arms.py (hunk 2 above)
Comment (LarsJanssenTUe) [reply robot_skills/src/robot_skills/arms.py]: > Why not use ros time to track time instead of approximating it this way?
Because we are not tracking time, we are computing the time!

File: robot_skills/src/robot_skills/arms.py (hunk 39 above)
[the remark this answers is not in the document]
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/arms.py]: The whole thing is fuzzy and no documentation yields a definitive answer if it's torque (Newton/meter) or force (Newton).
So: don't change at all...

File: robot_skills/src/robot_skills/util/entity.py (hunk 36 above)
Comment (ar13pit) [reply robot_skills/src/robot_skills/util/entity.py]: The two things are different. It shouldn't have been a dict in the first place. In the legacy message definition a field named `tagnames` was defined but never used. Only `tags` is used. Once the message definition in people_recognition is cleaned up then `tagnames` should be removed completely.

File: robot_skills/src/robot_skills/force_sensor.py (hunk 5 above)
Comment (LoyVanBeek) [reply robot_skills/src/robot_skills/force_sensor.py]: It uses a `Wrench` message which should adhere to https://www.ros.org/reps/rep-0103.html#units

File: robot_skills/src/robot_skills/arms.py (hunk 20 above)
Comment (LarsJanssenTUe) [reply robot_skills/src/robot_skills/arms.py]: This is fixed for now using a subclass as Loy proposed. However this should be fixed differently, see #911

File: robot_skills/src/robot_skills/arms.py [hunk 40]
```diff
@@ -113,10 +119,12 @@ def has_joint_trajectory(self, configuration):
         """
         return configuration in self._available_joint_trajectories
 
-    def send_joint_trajectory(self, configuration, timeout=5):
+    def send_joint_trajectory(self, configuration, timeout=5, max_joint_vel=0.7):
+        print()
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]: Why???????

File: robot_skills/src/robot_skills/arms.py [hunk 41]
```diff
@@ -432,7 +460,8 @@ def close(self):
         try:
             rospy.loginfo("{0} arm cancelling all goals on all arm-related ACs on close".format(self.side))
         except AttributeError:
-            print("{0} arm cancelling all goals on all arm-related ACs on close. rospy is already deleted.".format(self.side))
+            print("{0} arm cancelling all goals on all arm-related ACs on close. rospy is already deleted.".
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arms.py]:
```suggestion
            rospy.logerr("{0} arm cancelling all goals on all arm-related ACs on close. rospy is already deleted.".
```

File: robot_skills/src/robot_skills/base.py [hunk 42]
```diff
@@ -192,26 +198,35 @@ def move(self, position_constraint_string, frame):
...
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/base.py]: Do we want to get rid of the distance=speed*timeout?

File: robot_skills/src/robot_skills/robot.py [hunk 43]
```diff
@@ -231,6 +265,11 @@ def seq_or_none(obj):
                 discarded_reasons.append((arm_name, "required trajectories failed"))
                 continue
 
+            # Force sensor availability
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/robot.py]: Ugly to add it this way, create an issue to get rid of this.

File: robot_skills/src/robot_skills/util/entity.py (hunk 36 above)
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/util/entity.py]: Add ToDo or add issue

Comment (LarsJanssenTUe) [reply robot_skills/src/robot_skills/util/entity.py]: ToDo is added

File: robot_skills/src/robot_skills/base.py (hunk 42 above)
Comment (LarsJanssenTUe) [reply robot_skills/src/robot_skills/base.py]: By default no, but  if you want to make things hard on yourself by using the extra arguments I see it as extra freedom so I would say yes

File: robot_skills/src/robot_skills/torso.py (hunk 23 above)
Comment (LarsJanssenTUe) [reply robot_skills/src/robot_skills/torso.py]: Extra comment has been added to indicated that this is unfortunately not going to work unless the entire concept of joint_groups is kicked out

File: robot_skills/src/robot_skills/robot.py (hunk 43 above)
Comment (LarsJanssenTUe) [reply robot_skills/src/robot_skills/robot.py]: issure #914
